File: find-the-duplicate-number/find-the-duplicate-number.java
import java.util.*;

class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer,Integer> dict = new HashMap<>();
        
        for(int num: nums){
            if (dict.containsKey(Integer.valueOf(num))){
                return num;
            }
            dict.put(Integer.valueOf(num),1);
        }
        return -1;
    }
}

File: find-the-duplicate-number/find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        d = {}
        for i in nums:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
        
        for i in nums:
            count = d[i]
            if count > 1:
                return i